public/app/plugins/datasource/jaeger/components/QueryEditor.tsx

```diff
@@ -1,54 +1,78 @@
 import { css } from '@emotion/css';
 import { QueryEditorProps } from '@grafana/data';
 import { selectors } from '@grafana/e2e-selectors';
-import { InlineField, InlineFieldRow, Input, RadioButtonGroup } from '@grafana/ui';
+import { FileDropzone, InlineField, InlineFieldRow, Input, RadioButtonGroup, useTheme2 } from '@grafana/ui';
 import React from 'react';
 import { JaegerDatasource } from '../datasource';
 import { JaegerQuery, JaegerQueryType } from '../types';
 import { SearchForm } from './SearchForm';
 
 type Props = QueryEditorProps<JaegerDatasource, JaegerQuery>;
 
-export function QueryEditor({ datasource, query, onChange }: Props) {
+export function QueryEditor({ datasource, query, onChange, onRunQuery }: Props) {
+  const theme = useTheme2();
+
+  const renderEditorBody = () => {
+    switch (query.queryType) {
+      case 'search':
+        return <SearchForm datasource={datasource} query={query} onChange={onChange} />;
+      case 'upload':
+        return (
+          <div className={css({ padding: theme.spacing(2) })}>
+            <FileDropzone
+              options={{ accept: '.json', multiple: false }}
```

Comment:
Can we do without requiring the .json suffix? having a suffix of file type is not really necessary and for example on linux or mac a rarely add suffix to text docs. If you then try to upload such file it just does not do anything and does not give you any error message to suggest what is wrong.

Reply:
sure, I didn't know about this.